scripts/generate_site_sidebar/generate_site_sidebar.go

```diff
@@ -0,0 +1,201 @@
+    - [Ecosystem](guides/ecosystem/)
+        - [Kustomize](guides/ecosystem/kustomize/)
+        - [Helm](guides/ecosystem/helm/)
+        - [Open Application Model (OAM)](guides/ecosystem/oam/)
+- [Reference](reference/)
+    - [pkg](reference/pkg/)
+        - [diff](reference/pkg/diff/)
+        - [get](reference/pkg/get/)
+        - [init](reference/pkg/init/)
+        - [update](reference/pkg/update/)
+    - [fn](reference/fn/)
+        - [export](reference/fn/export/)
+        - [run](reference/fn/run/)
+        - [sink](reference/fn/sink/)
+        - [source](reference/fn/source/)
+    - [live](reference/live/)
+        - [apply](reference/live/apply/)
+        - [destroy](reference/live/destroy/)
+        - [diff](reference/live/diff/)
+        - [fetch-k8s-schema](reference/live/fetch-k8s-schema/)
+        - [init](reference/live/init/)
+        - [preview](reference/live/preview/)
+        - [status](reference/live/status/)
+- [Concepts](concepts/)
+    - [API Conventions](concepts/api-conventions/)
+    - [Architecture](concepts/architecture/)
+    - [Functions](concepts/functions/)
+    - [Packaging](concepts/packaging/)
+- [FAQ](faq/)
+- [Contact](contact/)`)
+}
+
+func printBookOutline() {
+	sourcePath := "site/book"
+	chapters := collectChapters(sourcePath)
+
+	printChapters(chapters)
+}
+
+func collectChapters(source string) []chapter {
+	chapters := make([]chapter, 0)
+	chapterDirs, err := ioutil.ReadDir(source)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+
+	for _, dir := range chapterDirs {
+		if dir.IsDir() {
+			chapters = append(chapters, getChapter(dir.Name(), filepath.Join(source, dir.Name())))
+		}
+	}
+
+	return chapters
+}
+
+func getChapter(chapterDirName string, chapterDirPath string) chapter {
+	chapterBuilder := chapter{}
+
+	// Split into chapter number and hyphenated name
+	splitDirName := strings.SplitN(chapterDirName, "-", 2)
+	chapterBuilder.Number = splitDirName[0]
+	chapterBuilder.Name = titlecase.Title(strings.ReplaceAll(splitDirName[1], "-", " "))
+
+	pageFiles, err := ioutil.ReadDir(chapterDirPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+
+	for _, pageFile := range pageFiles {
+		if filepath.Ext(pageFile.Name()) == markdownExtension {
+			chapterBuilder.Pages = append(chapterBuilder.Pages,
+				getPage(pageFile.Name(), chapterBuilder.Name, chapterDirPath))
+		}
+	}
+
+	return chapterBuilder
+}
+
+func getPage(pageFileName string, defaultName string, parentPath string) page {
+	// Split into page number and hyphenated name.
+	splitPageName := strings.SplitN(pageFileName, "-", 2)
+
+	pageName := defaultName
+	if pageFileName != introPage {
+		// Strip page number and extension from file name.
+		pageTitle := regexp.MustCompile(`^\d\d-?`).ReplaceAll([]byte(pageFileName), []byte(""))
+		pageName = titlecase.Title(strings.ReplaceAll(strings.ReplaceAll(string(pageTitle), ".md", ""), "-", " "))
```

Comment:
so is this logic duplicated in javascript and Go now? If you're preprocessing in Go, why do you need the runtime Javascript logic?

Reply:
Artifact of performing the logic at runtime first. Removed in favor of the preprocessing in Go.